firmware/application/ui/ui_btngrid.cpp

```diff
@@ -183,16 +192,26 @@ void BtnGridView::insert_item(const GridItem& new_item, size_t position, bool in
 }
 
 void BtnGridView::show_hide_arrows() {
+    portapack::async_tx_enabled = true;
     if (highlighted_item == 0) {
         set_arrow_up_enabled(false);
     } else {
         set_arrow_up_enabled(true);
     }
-    if (highlighted_item == (menu_items.size() - 1)) {
+    if (highlighted_item == (menu_items.size() - 1) || menu_items.size() < displayed_max) {
```

Comment:
Remove menu_items.size() < displayed_max from here, you are checking it just after